### PR TITLE
build: on-demand DKP-CLI docs for v2.1, v2.2

### PR DESCRIPTION
**Pull request by floridoo**
## Jira Ticket

https://d2iq.atlassian.net/browse/D2IQ-94468

## Description of changes being made
Restores the known, previously deactivated automation around the DKP CLI docs. The automation is not scheduled, but needs to be triggered manually when there is a need to sync docs.

### CODE REVIEW

**Review by cmastr — APPROVED** [review]

**Review by tillt — APPROVED** [review]
lgtm

**Review by mhrabovcin — APPROVED** [review]

**Comment by floridoo** [comment]
I am not authorized to merge here. Could someone with sufficient permissions please press merge? Thank you!

File: .github/workflows/dkp-cli.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Update dkp-cli docs
+on:
+  workflow_dispatch:
```
Comment (aweris) [inline .github/workflows/dkp-cli.yml:3]: [question] Are we planning to trigger this manually?

File: .github/workflows/dkp-cli.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        type: string
+        required: true
+        default: v2.3.1
+      target-version:
+        description: Target documentation version
+        type: choice
+        default: "2.3"
+        options:
+          - "2.3"
+          - "2.2"
```
Comment (aweris) [inline .github/workflows/dkp-cli.yml:16]: Maybe add `required: false` for consistency

File: .github/workflows/dkp-cli.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+jobs:
+  dkp-cli-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (aweris) [inline .github/workflows/dkp-cli.yml:22]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/dkp-cli.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        with:
+          fetch-depth: 0
+      - name: Update dkp-cli docs
+        run: |
+          ci/update-dkp-cli-docs.sh
+        env:
+          DKP_CLI_VERSION: ${{ inputs.dkp-version }}
+          TARGET_VERSION: ""
```
Comment (aweris) [inline .github/workflows/dkp-cli.yml:30]: Is this intentional? If it's true, you can remove the env var

**Review by aweris — COMMENTED** [review]
Can you lint steps a little bit as well? it's a little bit hard to read because there is no space between them.

File: .github/workflows/dkp-cli.yml (hunk 4 above)
Comment (floridoo) [reply .github/workflows/dkp-cli.yml]: No, this was not intentional, fixed.

File: .github/workflows/dkp-cli.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+name: Update dkp-cli docs
+on:
+  workflow_dispatch:
+    inputs:
+      dkp-version:
+        description: DKP version to generate the docs for, e.g. v2.3.1
+        type: string
+        required: true
+        default: v2.3.1
+      target-version:
```
[the remark this answers is not in the document]
Comment (floridoo) [reply .github/workflows/dkp-cli.yml:10]: It is now

**Pull request merged**